src/components/FolderListBar.jsx

```diff
@@ -1,24 +1,36 @@
-import style from "./FolderListBar.module.css";
+import styles from "./FolderListBar.module.css";
 import FolderButton from "./FolderButton";
 import addImg from "../images/add.svg";
-import { useState } from "react";
+import { useState, useMemo } from "react";
 import FolderOptions from "./FolderOptions";
+import AddFolderModal from "../modal/AddFolderModal/AddFolderModal";
+import EditModal from "../modal/EditModal/EditModal";
+import { MODALS } from "../modal/modals";
+import DeleteFolderModal from "../modal/DeleteFolderModal/DeleteFolderModal";
+import ShareModal from "../modal/ShareModal/ShareModal";
+import { copyClipBoard } from "../util/copyClipBoard";
+
+const WHOLE_BUTTON = {
+  id: 1,
+  created_at: "2023-06-04T20:59:39.293024+00:00",
+  name: "전체",
+  user_id: 1,
+  favorite: true,
+  link: {
+    count: 0,
+  },
+};
 
 function FolderListBar({ folderList, onClick }) {
-  const wholeBtn = {
-    id: 1,
-    created_at: "2023-06-04T20:59:39.293024+00:00",
-    name: "전체",
-    user_id: 1,
-    favorite: true,
-    link: {
-      count: 0,
-    },
-  };
-  const newFolderList = [wholeBtn, ...folderList];
-  const [nowFolderName, setNowFolderName] = useState("");
-  const [nowFolderId, setNowFolderId] = useState(1);
-  const [isClicked, setIsClicked] = useState(0);
+  const [currentFolderName, setCurrentFolderName] = useState("");
+  const [currentFolderId, setCurrentFolderId] = useState(1);
+  const [currentUserId, setCurrentUserId] = useState(null);
+  const [isModalClicked, setIsModalClicked] = useState({
+    add: false,
+    edit: false,
+    deleteFolder: false,
+    share: false,
+  });
```

Comment:
가능한 모달 전체 상태를 가지고 있기보다(이러면 모달하나 추가될 때마다 여기 들어와서  ~: false 추가해주어야함) currentClickedModal 이라고 해서 그냥 문자열을 받아도 될 거 같아요!
```suggestion
  const [currentClickedModal, setCurrentClickedModal] = useState('share');
```